### PR TITLE
[PW-2688] 3DS1 & 3DS2 payments - Handle the frontend

### PR DESCRIPTION
**Pull request by cyattilakiss**
## Tested scenarios

**Fixed issue**:

### CODE REVIEW

File: src/Handlers/CardsPaymentMethodHandler.php [hunk 1]
```diff
@@ -249,11 +257,19 @@ public function finalize(
...
+        /*$context = $salesChannelContext->getContext();
+        $this->orderTransactionStateHandler->paid($transaction->getOrderTransaction()->getId(), $context);*/
+
+        /*try {
             $this->resultHandler->processResult($transaction, $request, $salesChannelContext);
         } catch (PaymentException $exception) {
-            throw new AsyncPaymentFinalizeException($exception->getMessage());
-        }
+            throw new AsyncPaymentFinalizeException($transaction->getOrderTransaction()->getId(), $exception->getMessage());
+        }*/
```
Comment (acampos1916) [inline src/Handlers/CardsPaymentMethodHandler.php:272]: Should we remove these comments?

File: src/Handlers/PaymentResponseHandler.php [hunk 2]
```diff
@@ -208,21 +216,27 @@ public function handleShopwareApis(
                 // Tag order as paid
                 $this->transactionStateHandler->paid($orderTransactionId, $context);
                 break;
+            case self::REFUSED:
+                //Sync response, do nothing no, wait for finalize()
```
Comment (acampos1916) [inline src/Handlers/PaymentResponseHandler.php:220]:
```suggestion
                //Sync response, do nothing, wait for finalize()
```

File: src/Handlers/CardsPaymentMethodHandler.php (hunk 1 above)
Comment (cyattilakiss) [reply src/Handlers/CardsPaymentMethodHandler.php:274]: Let's keep it until we figure out how to test the 3DS1 flow and disabling the CSRF protection until it's fixed in shopware

File: src/Handlers/PaymentResponseHandler.php [hunk 3]
```diff
@@ -264,4 +278,32 @@ public function handleAdyenApis(
...
+     * @param SalesChannelContext $salesChannelContext
+     */
+    public function finalize(
+        AsyncPaymentTransactionStruct $transaction,
+        SalesChannelContext $salesChannelContext
+    ): void {
+        $orderTransactionId = $transaction->getOrderTransaction()->getId();
+        $context = $salesChannelContext->getContext();
+
+        $this->logger->debug($transaction->getOrderTransaction()->getCustomFields());
```
Comment (cyattilakiss) [inline src/Handlers/PaymentResponseHandler.php:294]: keep it here until we test the ERROR sync response

File: src/Handlers/PaymentResponseHandler.php [hunk 4]
```diff
@@ -264,4 +278,32 @@ public function handleAdyenApis(
...
+        switch ($resultCode) {
+            case self::AUTHORISED:
+                $this->transactionStateHandler->paid($orderTransactionId, $context);
+                break;
+            case self::REFUSED:
+            case self::ERROR:
+                $this->transactionStateHandler->fail($orderTransactionId, $context);
+                break;
+            default:
+                //
```
Comment (cyattilakiss) [inline src/Handlers/PaymentResponseHandler.php:306]:
```suggestion
                //TODO log unhandled result code
```

File: src/Resources/app/storefront/src/checkout/confirm-order.plugin.js [hunk 5]
```diff
@@ -71,15 +78,33 @@ export default class ConfirmOrderPlugin extends Plugin {
...
-    afterPaymetStatus(response) {
-        console.log(response);
-    }
+    handlePaymentAction(paymentAction) {
+        const paymentActionResponse = JSON.parse(paymentAction);
+
+        if (paymentActionResponse.isFinal === true) {
+            console.log(paymentActionResponse, window.returnUrl);
+            location.href = window.returnUrl;
+            return;
```
Comment (cyattilakiss) [inline src/Resources/app/storefront/src/checkout/confirm-order.plugin.js:97]:
```suggestion
            location.href = window.returnUrl;
```

File: src/Resources/views/storefront/page/checkout/confirm/index.html.twig [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{% sw_extends '@Storefront/storefront/page/checkout/confirm/index.html.twig' %}
+
+{% block base_footer %}
+    {{ parent() }}
+    <div class="modal" tabindex="-1" role="dialog" data-adyen-payment-action-modal>
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title">Verification</h5>
+                </div>
```
Comment (cyattilakiss) [inline src/Resources/views/storefront/page/checkout/confirm/index.html.twig:10]: Is the modal header mandatory?
We could remove it I guess.

```suggestion
```

File: src/Service/PaymentDetailsService.php [hunk 7]
```diff
@@ -146,12 +146,14 @@ public function doPaymentDetails(
         ];
 
         try {
-            $response = $this->checkoutService->paymentsDetails($request);
+            return $this->paymentResponseHandler->handlePaymentResponse(
+                $this->checkoutService->paymentsDetails($request),
+                $orderNumber,
+                $context
+            );
```
Comment (cyattilakiss) [inline src/Service/PaymentDetailsService.php:153]: Just to be a bit more readable

```suggestion
            $result = $this->checkoutService->paymentsDetails($request);
            
            return $this->paymentResponseHandler->handlePaymentResponse(
                $result,
                $orderNumber,
                $context
            );
```

**Pull request merged**